***** NOT A REAL PATCH FILE ******

i just made this to document how i changed the original date patch
changed
    n = (lines * bh) - 1;
        back to
    n = lines * bh;
        so i deleted that section here
changed
    drawdate(x, lines * bh, w);
        to
    drawdate(x, (lines + 1) * bh, w);
changed 
	mh = (lines + 1) * bh;
        to
	mh = (lines + 2) * bh;
        this was not in the original patch file, but the line was at 675 for my build, in void setup(void){}

diff --git a/dmenu.c b/dmenu.c
index 7cf253b..5428f67 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -143,6 +143,22 @@ drawitem(struct item *item, int x, int y, int w)
 	return drw_text(drw, x, y, w, bh, lrpad / 2, item->text, 0);
 }
 
+static int
+drawdate(int x, int y, int w)
+{
+    char date[128];
+    time_t t = time(NULL);
+    struct tm *tm = localtime(&t);
+
+    /* Hour:Minute DayOfTheWeek DayOfTheMonth Month Year */
+    strftime(date, sizeof(date), "%H:%M %A %d %B %Y", tm);
+
+	drw_setscheme(drw, scheme[SchemeSel]);
+
+	int r = drw_text(drw, x, y, w, bh, lrpad / 2, date, 0);
+	return r;
+}
+
 static void
 drawmenu(void)
 {
@@ -172,6 +188,8 @@ drawmenu(void)
 		/* draw vertical list */
 		for (item = curr; item != next; item = item->right)
 			drawitem(item, x, y += bh, mw - x);
+
+        drawdate(x, (lines + 1) * bh, w);
 	} else if (matches) {
 		/* draw horizontal list */
 		x += inputw;
-- 
2.41.0


	-mh = (lines + 1) * bh;
	+mh = (lines + 2) * bh;
